### ISSUE TITLE
[New rule] Prefer matching widget file name

### ISSUE THREAD

**Issue by incendial** [dart-code-checker/dart-code-metrics#391]
**Please describe what the rule should do:**
The rule should recommend naming a widget and a file that contains the widget in the same way. Given the widget name `SomeWidget` then the containing file should be named as `some_widget.dart`.

**If your rule is inspired by other please provide link to it:**

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

Bad:

**some_widget.dart**
```dart
// LINT
class someOtherWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    ...
  }
}
```

Good:

**some_widget.dart**
```dart
class someWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    ...
  }
}
```

**Comment by incendial** [comment]
Available in 4.2.0 release.
